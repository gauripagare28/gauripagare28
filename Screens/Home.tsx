import React, {useState, useEffect} from 'react';
import {View, TouchableOpacity, Text,  FlatList,StyleSheet, ScrollView} from 'react-native';
import {TextInput,Button} from "react-native-paper"
//import fetch from 'node-fetch';
interface Iprops{
    navigation:any,route:any
}
const Home:React.FC<Iprops> = ({navigation}) => {
    const [flag, setFlag] = useState(true);

const [searchData, setSearchData] = useState<any>('');
const [newsData, setNewsData] = useState<any>([]);
const [endReached, setEndReached] = useState<any>(true);
const [filterdata,setFilterData]=useState<any>([]);
const [page,setPage]=useState(0);

useEffect(() => {
       getData();
        setInterval(()=>getData(), 10000)
       
  });

const getData = () => {
    
        let url=`https://hn.algolia.com/api/v1/search_by_date?tags=story&page=${page}`;
        fetch( url )
                 .then(response => response.json())
                 .then((responseJson) => {
                 const ndata : any = page==0 ? responseJson.hits : [...newsData, ...responseJson.hits];
                 setNewsData(ndata);
                 setPage(page+1);
                 console.log(page)
                })
               
                
};


// const updateData = () => {
//                 const pageUpdate:number=page + 1
            
//         let url= `https://hn.algolia.com/api/v1/search_by_date?tags=story&page=${pageUpdate}`
//         fetch( url )
//                 .then(response => response.json())
//                 .then((responseJson) => {
//                 const ndata:any = [...newsData, ...responseJson.hits];
//                 console.log(">>>>>>>>>>>>>>>",ndata)
//                 setNewsData(ndata);
//                 })
//                 .catch(()=>console.log("data not found"))  
//   };

const searchFilter = (text:any) => {

        setEndReached(false)
        const sfilter = newsData.filter((ele:any) => {
                 return (
                         ele.author.toLowerCase().includes(text.toLowerCase()) ||
                         ele.title.toLowerCase().includes(text.toLowerCase())
                        );
        });
      
      setFilterData(sfilter)
  };

const searchByDate = () => {
    setFlag(false)
        const sdate = newsData
        sdate.sort((a:any, b:any):any => (a.created_at > b.created_at ? 1 : -1));
        setNewsData(sdate);
};

const searchByTitle = () => {
    setFlag(false)
        const stitle = newsData;
        stitle.sort((a:any, b:any) => (a.title > b.title ? 1 : -1));
        setNewsData(stitle);
};

const renderItem = ({item}:any) => {
    return (
            
                <View style={{flex:1,alignItems:"center",paddingBottom:10}}>
                    <TouchableOpacity
                         onPress={() => navigation.navigate('Info', {rawjson: item})}
                         style={styles.buttoninfo}>
                        <Text><Text style={{fontWeight: 'bold'}}>Title : </Text>{item.title}</Text>
                            <Text>
                            <Text style={{fontWeight: 'bold'}}>URL : </Text>
                            {item.url}
                        </Text>
                        <Text>
                            <Text style={{fontWeight: 'bold'}}>created_at : </Text>
                            {item.created_at}
                        </Text>
                        <Text>
                            <Text style={{fontWeight: 'bold'}}>author : </Text> {item.author}
                        </Text>
                    </TouchableOpacity>
                    </View>
    
    );
  };

  return (
    
    <View style={{flex:1,}}>
        
        <TextInput
        mode="outlined"
       style={{margin:10}}
      label="Search by Author or title"
      value={searchData}
      onChangeText={text => setSearchData(text)}
    />
        
        
            <Button
                style={{margin:10}}
                mode="contained"
                onPress={() => searchFilter(searchData)}
                disabled={searchData==="" ? true : false}>
               Search
            </Button>
            <Button
                mode="contained"
                style={{margin:10}}
                onPress={searchByDate}
                >Filter by Created_At
            </Button>
            <Button
                mode="contained"
                style={{margin:10}}
                onPress={searchByTitle}
               >Filter by Title
            </Button>

     
      {searchData.length>1
            ? 
            flag ?
            <FlatList
                         data={filterdata}
                         renderItem={item => renderItem(item)}
                        keyExtractor={(index, item) => item.toString()} />
                        :
                        <FlatList
                        data={filterdata}
                        renderItem={item => renderItem(item)}
                       keyExtractor={(index, item) => item.toString()} />
               

            : <FlatList
                        data={newsData}
                        renderItem={item => renderItem(item)}
                        keyExtractor={(index, item) => item.toString()}
                       // onEndReached={endReached ? ()=>updateData() : null}
                        // onEndReachedThreshold={0.03} 
                        />
        }
</View>

  );
};

const styles = StyleSheet.create(
{

    
buttoninfo:{
            width: 350,
            borderWidth: 1,
            borderColor: '#cccc',
            padding: 16,
            margin:8,
           }
}
)

export default Home;
