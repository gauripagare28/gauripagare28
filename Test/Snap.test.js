import React  from "react"
import "react-native"
import renderer from "react-test-renderer"
import Home from "../Screens/Home"

it('Snapshot test', () => {
            const result= renderer.create(<Home />).toJSON();
            expect(result).toMatchSnapshot();
  });