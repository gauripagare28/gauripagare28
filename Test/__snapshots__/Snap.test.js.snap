// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "margin": 10,
      }
    }
  >
    <View>
      <View
        pointerEvents="none"
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 6,
            },
            Object {
              "backgroundColor": "#f6f6f6",
              "borderColor": "rgba(0, 0, 0, 0.54)",
              "borderRadius": 4,
              "borderWidth": 1,
            },
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "paddingBottom": 0,
            },
            Object {
              "minHeight": 64,
              "paddingTop": 8,
            },
          ]
        }
      >
        <View
          pointerEvents="none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "backgroundColor": "#f6f6f6",
                "bottom": 4,
                "left": 10,
                "opacity": 0,
                "position": "absolute",
                "right": 0,
                "top": 6,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                ],
                "width": 8,
              }
            }
          />
          <Text
            numberOfLines={1}
            style={
              Object {
                "backgroundColor": "#f6f6f6",
                "color": "transparent",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "left": 18,
                "opacity": 0,
                "paddingHorizontal": 0,
                "position": "absolute",
                "textAlign": "left",
                "top": 37,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                  Object {
                    "translateY": 0,
                  },
                  Object {
                    "scale": 1,
                  },
                  Object {
                    "scaleY": 1,
                  },
                ],
                "writingDirection": "ltr",
              }
            }
          >
            Search by Author or title
          </Text>
          <Text
            numberOfLines={1}
            onLayout={[Function]}
            style={
              Object {
                "color": "#6200ee",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "left": 0,
                "opacity": 0,
                "paddingHorizontal": 14,
                "position": "absolute",
                "textAlign": "left",
                "top": 36,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                  Object {
                    "translateY": 0,
                  },
                  Object {
                    "scale": 1,
                  },
                ],
                "writingDirection": "ltr",
              }
            }
          >
            Search by Author or title
          </Text>
          <Text
            numberOfLines={1}
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.54)",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "left": 0,
                "opacity": 0,
                "paddingHorizontal": 14,
                "position": "absolute",
                "textAlign": "left",
                "top": 36,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                  Object {
                    "translateY": 0,
                  },
                  Object {
                    "scale": 1,
                  },
                ],
                "writingDirection": "ltr",
              }
            }
          >
            Search by Author or title
          </Text>
        </View>
        <TextInput
          allowFontScaling={true}
          editable={true}
          multiline={false}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholder=""
          placeholderTextColor="rgba(0, 0, 0, 0.54)"
          rejectResponderTermination={true}
          selectionColor="#6200ee"
          style={
            Array [
              Object {
                "flexGrow": 1,
                "margin": 0,
                "paddingHorizontal": 14,
                "zIndex": 1,
              },
              Object {
                "height": 56,
              },
              Object {
                "paddingBottom": 0,
                "paddingTop": 0,
              },
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "textAlign": "left",
                "textAlignVertical": "center",
              },
              false,
              Array [
                Object {},
              ],
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.12)",
        "borderColor": "transparent",
        "borderRadius": 4,
        "borderStyle": "solid",
        "borderWidth": 0,
        "elevation": 0,
        "margin": 10,
        "minWidth": 64,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": true,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 4,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <Text
          numberOfLines={1}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "textAlign": "left",
              },
              Array [
                Object {
                  "letterSpacing": 1,
                  "marginHorizontal": 16,
                  "marginVertical": 9,
                  "textAlign": "center",
                },
                undefined,
                Object {
                  "textTransform": "uppercase",
                },
                Object {
                  "color": "rgba(0, 0, 0, 0.32)",
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                Object {
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                undefined,
              ],
            ]
          }
        >
          Search
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#6200ee",
        "borderColor": "transparent",
        "borderRadius": 4,
        "borderStyle": "solid",
        "borderWidth": 0,
        "elevation": 2,
        "margin": 10,
        "minWidth": 64,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 0.75,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 1.5,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": undefined,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 4,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <Text
          numberOfLines={1}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "textAlign": "left",
              },
              Array [
                Object {
                  "letterSpacing": 1,
                  "marginHorizontal": 16,
                  "marginVertical": 9,
                  "textAlign": "center",
                },
                undefined,
                Object {
                  "textTransform": "uppercase",
                },
                Object {
                  "color": "#ffffff",
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                Object {
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                undefined,
              ],
            ]
          }
        >
          Filter by Created_At
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#6200ee",
        "borderColor": "transparent",
        "borderRadius": 4,
        "borderStyle": "solid",
        "borderWidth": 0,
        "elevation": 2,
        "margin": 10,
        "minWidth": 64,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 0.75,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 1.5,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": undefined,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 4,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <Text
          numberOfLines={1}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "textAlign": "left",
              },
              Array [
                Object {
                  "letterSpacing": 1,
                  "marginHorizontal": 16,
                  "marginVertical": 9,
                  "textAlign": "center",
                },
                undefined,
                Object {
                  "textTransform": "uppercase",
                },
                Object {
                  "color": "#ffffff",
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                Object {
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                undefined,
              ],
            ]
          }
        >
          Filter by Title
        </Text>
      </View>
    </View>
  </View>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;
